pages/p/[id].tsx

```diff
@@ -5,7 +5,7 @@ import Layout from "../../components/Layout"
 import { PostProps } from "../../components/Post"
 
 export const getServerSideProps: GetServerSideProps = async ({ params }) => {
-  const post = await prisma.post.findUnique({
+  const post = await prisma.post.findUniqueOrThrow({
```

Comment:
Changed due to type error coming from `getServerSideProps` cannot return null. Changed to `findUniqueOrThrow` to always return post.